<!-- Visual -->
<element name="visual" required="*">
  <attribute name="name" type="string" default="__default__" required="0"/>
  <attribute name="cast_shadows" type="bool" default="true" required="0"/>
  <attribute name="laser_retro" type="double" default="0.0" required="0"/>
  <attribute name="transparency" type="double" default="0.0" required="0"/>
  
  <element name="origin" required="0">
    <attribute name="pose" type="pose" default="0 0 0 0 0 0" required="1"/>
  </element> <!-- End Origin -->

  <element name="material" required="0">
    <attribute name="script" type="string" default="__default__" required="0"/>
    <attribute name="normal_map" type="string" default="__default__" required="0"/>
    <element name="color" required="0">
      <attribute name="rgba" type="color" default="0 0 0 1" required="1"/>
    </element> <!-- End Color -->
  </element> <!-- End Material -->

  <include filename="geometry.sdf" required="1"/>
</element> <!-- End Visual -->
